business/namespaces.go

```diff
@@ -613,6 +300,10 @@ func (in *NamespaceService) GetClusterNamespace(ctx context.Context, namespace s
 		log.Errorf("Unable to refresh cache for cluster [%s]: %s", cluster, err)
 	}
 
+	if !in.isAccessibleNamespace(result) {
+		return nil, &AccessibleNamespaceError{msg: "Namespace [" + namespace + "] is not accessible for Kiali"}
+	}
```

Comment:
Don't you want to do this access check above where the previous ones were done? `GetClusterNamespaces` might include it in the cache which would then bypass the access check the next time?

Reply:
After looking at this deeper, yeah, move it back up. Best to do the isAccessibleNamespace check (which is fast) before bothering to do all that client/cache lookup stuff.  This is done.